### PR TITLE
Update The Documentation

### LINKED ISSUE #129: Update the documentation of the project
Ensure that:
- The read me is updated with new integration instructions as the API has changed (Include android specific intergration instructions if applicable)
- Update java docs in the code to reflect the changes
- Update the screenshots/demo. Make sure you add a video for both iOS and android, as well as screenshots in the art folder
- Include known limitations/issues eg drawing under the control menu

### PR DESCRIPTION
**Pull request by emmanuelmuturia**
Fixes #129

Revamps the Read me and updates the documentation of the SDK to reflect KMP changes

### CODE REVIEW

**Comment by Fbada006** [comment]
Hey @emmanuelmuturia , I will hijack this and make some more changes (such as the iOS stuff) then merge.

**Comment by emmanuelmuturia** [comment]
> Hey @emmanuelmuturia , I will hijack this and make some more changes (such as the iOS stuff) then merge.

No worries...

**Comment by Fbada006** [comment]
Hey @CalebKL and @emmanuelmuturia , please go through this. As you can see, I have made several changes

File: artmaker/src/commonMain/kotlin/io/fbada006/artmaker/utils/ExportHelper.kt [hunk 1]
```diff
@@ -17,4 +17,9 @@ package io.fbada006.artmaker.utils
 
 import androidx.compose.ui.graphics.ImageBitmap
 
-expect suspend fun shareImage(imageBitmap: ImageBitmap?)
+/**
+ * Called to trigger sharing an image depending on the platform
+ *
+ * @param imageBitmap is the image to share
+ */
```
Comment (Fbada006) [inline artmaker/src/commonMain/kotlin/io/fbada006/artmaker/utils/ExportHelper.kt:24]: So as you can see @emmanuelmuturia, the approach I have taken is only providing docs to the expect calls. We can think of them as interfaces. As such, we only need to document the base method and not the implementations.

File: artmaker/src/commonMain/kotlin/io/fbada006/artmaker/customcolorpalette/SelectorIndicator.kt [hunk 2]
```diff
@@ -22,6 +22,11 @@ import androidx.compose.ui.graphics.drawscope.DrawScope
 import androidx.compose.ui.graphics.drawscope.Stroke
 import androidx.compose.ui.unit.dp
 
+/**
+ * Vertically draw the Selector Indicator.
+ *
+ * @param amount Represents the value from which the Half Indicator Thickness will be subtracted from to get the Selector Indicator's [Offset].
```
Comment (Fbada006) [inline artmaker/src/commonMain/kotlin/io/fbada006/artmaker/customcolorpalette/SelectorIndicator.kt:28]: You will also notice, @emmanuelmuturia , that I have removed all the names of the methods included in the documentation of the method. That is a bit like defining a word using the word itself.

**Review by Fbada006 — APPROVED** [review]
Fine with me but I will wait for an extra set of eyes before merging.

File: artmaker/src/commonMain/kotlin/io/fbada006/artmaker/composables/ArtMakerControlMenu.kt [hunk 3]
```diff
@@ -70,23 +70,32 @@ import io.fbada006.artmaker.utils.createPicker
...
+ * @param state
+ * @param onAction called during a user action
+ * @param onDrawEvent called when a drawing event happens
+ * @param onShowStrokeWidthPopup called when stroke settings button is clicked
+ * @param setBackgroundImage called to change the background image of the canvas
+ * @param imageBitmap is the background image, if any
+ * @param configuration global config
+ * @param onActivateEraser called when eraser button is clicked
+ * @param isEraserActive whether in erasing mode or not
+ * @param modifier is the modifier
```
Comment (CalebKL) [inline artmaker/src/commonMain/kotlin/io/fbada006/artmaker/composables/ArtMakerControlMenu.kt:84]: we could change the description to the modifier to something like
"optional param to set the layout of the Control Menu"

**Review by CalebKL — CHANGES_REQUESTED** [review]
@Fbada006 and @emmanuelmuturia, the documentation looks good to me. I only have two pointers. 
1. `@param modifier is the [Modifier]` We could not be too generic here and go with something like, `@param modifier is an optional param to set the layout of the control menu`
2. Since we support IOS and Android now, I think a KMP library can also support Native IOS. We could include this on the README.md. I was going through this [article](https://medium.com/@jvondermarck/developing-with-kotlin-compose-multiplatform-and-integrating-native-ios-libraries-a7b6b795b7ec) and I think its possible

**Comment by Fbada006** [comment]
The documentation has been updated as requested. As for working natively with iOS, we need to have extra steps to publish the lib to CocoPods, which has not been covered yet as it was not in the original scope of work. Should we go there, then we can update the documentation. An example of how to do that: https://www.marcogomiero.com/posts/2021/kmp-xcframework-official-support/

I will merge this now

**Pull request merged**